api/v1beta1/azuremachine_validation.go

```diff
@@ -124,9 +126,19 @@ func ValidateSystemAssignedIdentity(identityType VMIdentity, oldIdentity, newIde
 func ValidateUserAssignedIdentity(identityType VMIdentity, userAssignedIdentities []UserAssignedIdentity, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if identityType == VMIdentityUserAssigned && len(userAssignedIdentities) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath, "must be specified for the 'UserAssigned' identity type"))
+	if identityType == VMIdentityUserAssigned {
+		if len(userAssignedIdentities) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath, "must be specified for the 'UserAssigned' identity type"))
+		}
+		for _, identity := range userAssignedIdentities {
+			if identity.ProviderID != "" {
+				if _, err := arm.ParseResourceID(strings.TrimPrefix(identity.ProviderID, "azure://")); err != nil {
```

Comment:
We can't use the `azure.ParseResourceID()` func added in #3616 because it would cause a circular import error.

Reply:
The azure.go file itself doesn't import the `api` package, I wonder if we should move all azure "utils" functions to its own package so we can use the same logic in both places

Reply:
@mboersma thoughts on this?